hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -568,62 +567,20 @@ def dictionaries(
     ).map(dict_class)
 
 
-# See https://en.wikipedia.org/wiki/Unicode_character_property#General_Category
-CategoryName: "TypeAlias" = Literal[
-    "L",  #  Letter
-    "Lu",  # Letter, uppercase
-    "Ll",  # Letter, lowercase
-    "Lt",  # Letter, titlecase
-    "Lm",  # Letter, modifier
-    "Lo",  # Letter, other
-    "M",  #  Mark
-    "Mn",  # Mark, nonspacing
-    "Mc",  # Mark, spacing combining
-    "Me",  # Mark, enclosing
-    "N",  #  Number
-    "Nd",  # Number, decimal digit
-    "Nl",  # Number, letter
-    "No",  # Number, other
-    "P",  #  Punctuation
-    "Pc",  # Punctuation, connector
-    "Pd",  # Punctuation, dash
-    "Ps",  # Punctuation, open
-    "Pe",  # Punctuation, close
-    "Pi",  # Punctuation, initial quote
-    "Pf",  # Punctuation, final quote
-    "Po",  # Punctuation, other
-    "S",  #  Symbol
-    "Sm",  # Symbol, math
-    "Sc",  # Symbol, currency
-    "Sk",  # Symbol, modifier
-    "So",  # Symbol, other
-    "Z",  #  Separator
-    "Zs",  # Separator, space
-    "Zl",  # Separator, line
-    "Zp",  # Separator, paragraph
-    "C",  #  Other
-    "Cc",  # Other, control
-    "Cf",  # Other, format
-    "Cs",  # Other, surrogate
-    "Co",  # Other, private use
-    "Cn",  # Other, not assigned
-]
-
-
 @cacheable
 @defines_strategy(force_reusable_values=True)
 def characters(
     *,
     codec: Optional[str] = None,
     min_codepoint: Optional[int] = None,
     max_codepoint: Optional[int] = None,
-    categories: Optional[Collection[CategoryName]] = None,
-    exclude_categories: Optional[Collection[CategoryName]] = None,
+    categories: Optional[Categories] = None,
```

Comment:
notably this is a loosening of the type, from `Collection[CategoryName]` to `Iterable[CategoryName]`.

Reply:
Something something hyrum's Law / may as well type what works? I think typing as `Iterable` doesn't push the footgun up to the user, so to speak – more like `Collection` forces them to "fix" what in our determination is a "footgun" (but maybe they have a valid-but-cursed custom iterable type that doesn't implement e.g. `__contains__`).

But I am very unexperienced in maintaining forward-facing types for users, so I'm happy to defer this decision.

Reply:
Let's stick with the status quo (`Collection`) here, and get the rest in 🙂